src/st/np3/394F0.c

```diff
@@ -1,5 +1,514 @@
+                            yCoord2 /= 2;
+                        }
+                        for (prim = &g_PrimBuf[g_unkGraphicsStruct.D_800973F8];
+                             prim != NULL; prim = prim->next) {
+                            if (prim->drawMode == DRAW_HIDE) {
+                                prim->clut = 0x199;
+                                temp_rand = (Random() & 7) - 13;
+                                prim->x0 = prim->x2 = xCoord + temp_rand - 3;
+                                prim->x1 = prim->x3 = prim->x0 + 0x20;
+                                temp_rand = (Random() & 7) - 10;
+                                prim->y0 = prim->y1 = yCoord2 + temp_rand - 3;
+                                prim->y2 = prim->y3 = prim->y0 + 0x20;
+                                prim->p1 = 0;
+                                if (iterEnt2->zPriority > iterEnt1->zPriority) {
+                                    prim->priority = iterEnt2->zPriority + 1;
+                                } else {
+                                    prim->priority = iterEnt1->zPriority + 1;
+                                }
+                                prim->drawMode = DRAW_UNK02;
+                                break;
+                            }
+                        }
+                    }
+                    if ((iterEnt2->attack) &&
+                        (entFrom5C->hitPoints != 0x7FFF)) {
+                        miscVar1 = g_api.DealDamage(iterEnt1, iterEnt2);
+                        if (iterEnt2->hitboxState == 4) {
+                            miscVar1 = 0;
+                        }
+                        if ((g_Status.relics[RELIC_SPIRIT_ORB] & 2) &&
+                            !(entFrom5C->flags & FLAG_UNK_04000000) &&
+                            miscVar1) {
+                            otherEntity =
+                                AllocEntity(&g_Entities[224], &g_Entities[256]);
+                            if (otherEntity != NULL) {
+                                DestroyEntity(otherEntity);
+                                otherEntity->entityId = 4;
+                                otherEntity->pfnUpdate = EntityDamageDisplay;
+                                otherEntity->posX.i.hi = xCoord;
+                                otherEntity->posY.i.hi = yCoord2;
+                                otherEntity->params = miscVar1;
+                            }
+                        }
+                    } else {
+                        miscVar1 = 0;
+                    }
+                    if (miscVar1 != 0xC000) {
+                        if (miscVar1 & 0x8000) {
+                            miscVar2 = 9;
+                        } else {
+                            miscVar3 = (u16)iterEnt2->attackElement;
+                            // includes all elements 0x40 and up
+                            if (miscVar3 & 0xFFC0) {
+                                for (i = 0; i < 10; i++) {
```

Comment:
```suggestion
                                for (i = 0; i < LEN(D_801819AC); i++) {
```
but you would need to set the array size for `D_801819AC` first